src/main/java/com/ays/user/service/impl/UserServiceImpl.java

```diff
@@ -16,12 +16,10 @@
 import lombok.RequiredArgsConstructor;
 import org.springframework.data.domain.Page;
 import org.springframework.stereotype.Service;
-import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
 
 @Service
-@Transactional(readOnly = true)
```

Comment:
Since we are not only performing read operations but also handling saving operations within this service, adding `@Transactional(readOnly = true)` may not be the correct approach. Adding the `@Transactional` annotation would not serve any specific purpose for this service as there are no interdependent repository operations.